### ISSUE TITLE
[1.10.2] Server Crash when Environmental Controller Attempts to Apply Buffs

### ISSUE THREAD

**Issue by praetoras** [McJty/RFTools#706]
I just recently updated my modpack to the most current versions (details below) of RFTools and Forge. However, when trying to load the server, I found that whenever the environmental controller attempts to apply the buffs, the server crashes. I even edited my player's location away from the EC, and logged in without issue. But as soon as I entered the EC's area of influence, the server crashed again. This happened multiple times. Is this a conflict with the most recent version of forge? I'll be happy to roll that back until this is resolved.

rftools-1.10-5.19
Forge 12.18.1.2073
Minecraft Version: 1.10.2

**Crash Log:**

```
Description: Ticking block entity

java.lang.NullPointerException: Ticking block entity
	at mcjty.rftools.playerprops.BuffProperties.addBuffToPlayer(BuffProperties.java:125)
	at mcjty.rftools.blocks.environmental.modules.PotionEffectModule.processPlayers(PotionEffectModule.java:81)
	at mcjty.rftools.blocks.environmental.modules.PotionEffectModule.tick(PotionEffectModule.java:55)
	at mcjty.rftools.blocks.environmental.EnvironmentalControllerTileEntity.checkStateServer(EnvironmentalControllerTileEntity.java:317)
	at mcjty.rftools.blocks.environmental.EnvironmentalControllerTileEntity.func_73660_a(EnvironmentalControllerTileEntity.java:286)
	at net.minecraft.world.World.func_72939_s(World.java:1804)
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:619)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:709)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:387)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471)
	at java.lang.Thread.run(Unknown Source)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at mcjty.rftools.playerprops.BuffProperties.addBuffToPlayer(BuffProperties.java:125)
	at mcjty.rftools.blocks.environmental.modules.PotionEffectModule.processPlayers(PotionEffectModule.java:81)
	at mcjty.rftools.blocks.environmental.modules.PotionEffectModule.tick(PotionEffectModule.java:55)
	at mcjty.rftools.blocks.environmental.EnvironmentalControllerTileEntity.checkStateServer(EnvironmentalControllerTileEntity.java:317)
	at mcjty.rftools.blocks.environmental.EnvironmentalControllerTileEntity.func_73660_a(EnvironmentalControllerTileEntity.java:286)

-- Block entity being ticked --
Details:
	Name: rftools_environmental_controller // mcjty.rftools.blocks.environmental.EnvironmentalControllerTileEntity
	Block type: ID #2539 (tile.rftools.environmental_controller // mcjty.rftools.blocks.environmental.EnvironmentalControllerBlock)
	Block data value: 0 / 0x0 / 0b0000
	Block location: World: (-4699,77,1521), Chunk: (at 5,4,1 in -294,95; contains blocks -4704,0,1520 to -4689,255,1535), Region: (-10,2; contains chunks -320,64 to -289,95, blocks -5120,0,1024 to -4609,255,1535)
	Actual block type: ID #2539 (tile.rftools.environmental_controller // mcjty.rftools.blocks.environmental.EnvironmentalControllerBlock)
	Actual block data value: 0 / 0x0 / 0b0000
Stacktrace:
	at net.minecraft.world.World.func_72939_s(World.java:1804)
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:619)

-- Affected level --
Details:
	Level name: world
	All players: 1 total; [EntityPlayerMP['Starfall_Leyline'/264, l='world', x=-4694.05, y=74.00, z=1521.43]]
	Chunk stats: ServerChunkCache: 574 Drop: 0
	Level seed: -4733961995790873640
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (0,0,0), Chunk: (at 0,0,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 0 game time, 0 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
Stacktrace:
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:709)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:387)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471)
	at java.lang.Thread.run(Unknown Source)

-- System Details --
Details:
	Minecraft Version: 1.10.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_101, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 2149856864 bytes (2050 MB) / 2957508608 bytes (2820 MB) up to 2957508608 bytes (2820 MB)
	JVM Flags: 2 total; -Xms3084M -Xmx3084M
	IntCache: cache: 14, tcache: 0, allocated: 4, tallocated: 105
	FML: MCP 9.32 Powered by Forge 12.18.1.2073 91 mods loaded, 91 mods active

```

**Comment by McJty** [comment]
I'll check this out asap

**Comment by McJty** [comment]
Fixed in 5.20

**Comment by praetoras** [comment]
Thanks so much! will confirm working once I update to the new version :)
